app/controllers/users/folders/binaries/comment_likes_controller.rb

```diff
@@ -1,15 +1,23 @@
 class Users::Folders::Binaries::CommentLikesController < ApplicationController
   def create
     if current_user
-      binary = Binary.find_by(name: params[:binary_name])
-      user = binary.folder.owner
-      comment = binary.comments.find_by(id: params[:comment_id])
-      like = comment.likes.create(user: user)
-      redirect_to(binary.url)
+      @binary = Binary.find_by(name: params[:binary_name])
+      comment = @binary.comments.find_by(id: params[:comment_id])
+      comment.likes.create(user: current_user)
+      redirect_to(@binary.url)
     else
-      binary = Binary.find_by(name: params[:binary_name])
+      @binary = Binary.find_by(name: params[:binary_name])
       flash[:danger] = "Log in to add comments and likes."
-      redirect_to public_folders_binary_path(binary)
+      redirect_to public_folders_binary_path(@binary)
+    end
+  end
+
+  def delete
+    if current_user
+      @binary = Binary.find_by(name: params[:binary_name])
+      @comment = @binary.comments.find_by(id: params[:comment_id])
+      @comment.likes.find_by(user_id: current_user.id).destroy
```

Comment:
Do we need to find the binary here? Wonder if we could do something like:

`current_user.likes.find_by(comment_id: params[:comment_id]).destroy`

This would decrease your SQL queries from 3 calls to 1.